ogr/ogrsf_frmts/arrow/ogrfeatherwriterlayer.cpp

```diff
@@ -0,0 +1,435 @@
+                    pszType =  "GeometryCollection";
+                oColumn.Add("gdal:geometry_type", pszType);
+            }
+        }
+
+        auto kvMetadata = m_poSchema->metadata() ? m_poSchema->metadata()->Copy() :
+                              std::make_shared<arrow::KeyValueMetadata>();
+        kvMetadata->Append("geo", oRoot.Format(CPLJSONObject::PrettyFormat::Plain));
+        m_poSchema = m_poSchema->WithMetadata(kvMetadata);
+        CPLAssert(m_poSchema);
+    }
+}
+
+/************************************************************************/
+/*                          CreateWriter()                              */
+/************************************************************************/
+
+void OGRFeatherWriterLayer::CreateWriter()
+{
+    CPLAssert( m_poFileWriter == nullptr );
+
+    if( m_poSchema == nullptr )
+    {
+        CreateSchema();
+    }
+    else
+    {
+        FinalizeSchema();
+    }
+
+    auto options = arrow::ipc::IpcWriteOptions::Defaults();
+    options.memory_pool = m_poMemoryPool;
+
+    {
+        auto result = arrow::util::Codec::Create(m_eCompression);
+        if( !result.ok() )
+        {
+            CPLError(CE_Failure, CPLE_AppDefined,
+                     "Codec::Create() failed with %s",
+                     result.status().message().c_str());
+        }
+        else
+        {
+            options.codec.reset(result->release());
+        }
+    }
+
+    if( m_bStreamFormat )
+    {
+        auto result = arrow::ipc::MakeStreamWriter(m_poOutputStream,
+                                                   m_poSchema,
+                                                   options);
+        if( !result.ok() )
+        {
+            CPLError(CE_Failure, CPLE_AppDefined,
+                     "arrow::ipc::MakeStreamWriter() failed with %s",
+                     result.status().message().c_str());
+        }
+        else
+        {
+            m_poFileWriter = *result;
+        }
+    }
+    else
+    {
+        m_poFooterKeyValueMetadata = std::make_shared<arrow::KeyValueMetadata>();
+        auto result = arrow::ipc::MakeFileWriter(m_poOutputStream,
+                                                 m_poSchema,
+                                                 options,
+                                                 m_poFooterKeyValueMetadata);
+        if( !result.ok() )
+        {
+            CPLError(CE_Failure, CPLE_AppDefined,
+                     "arrow::ipc::MakeFileWriter() failed with %s",
+                     result.status().message().c_str());
+        }
+        else
+        {
+            m_poFileWriter = *result;
+        }
+    }
+}
+
+/************************************************************************/
+/*               DoSomethingBeforeFinalFlushGroup()                     */
+/************************************************************************/
+
+void OGRFeatherWriterLayer::DoSomethingBeforeFinalFlushGroup()
```

Comment:
Is there anymore explanation you could add about what "something" is? :laughing:

Reply:
existing comment inside the function moved above. The name of the method is admitedly not super clear but this is a virtual method implemented by the Parquet and Arrow drivers, and they do slightly different things...

Reply:
> The name of the method is admitedly not super clear but this is a virtual method implemented by the Parquet and Arrow drivers, and they do slightly different things...

In that case would something like "PerformStepsBeforeFinalFlushGroup" be a little better?

Reply:
done

Reply:
Thanks!